### PR TITLE
Add service to mark envelopes as consumed

### PR DESCRIPTION
**Pull request by rkondratowicz**
The service will be used by a new controller action added in next PR.

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeUpdateServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    public void markAsConsumed_should_throw_an_exception_if_envelope_with_given_id_does_not_exist() throws Exception {
+        //given
+        given(repo.findById(any())).willReturn(Optional.empty());
+
+        // when
+        Throwable exc = catchThrowable(() -> service.markAsConsumed(randomUUID(), "some_service"));
+
+        // then
+        assertThat(exc)
+            .isNotNull()
```
Comment (luigibk) [inline src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeUpdateServiceTest.java]: isNotNull() looks redundant

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeUpdateService.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        Envelope envelope =
+            repo
+                .findById(envelopeId)
+                .orElseThrow(() -> new EnvelopeNotFoundException());
+
+        accessService.assertCanUpdate(envelope.getJurisdiction(), serviceName);
+
+        envelope.setStatus(Event.DOC_CONSUMED);
+        repo.saveAndFlush(envelope);
+    }
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeUpdateService.java]: Should we also record `ProcessEvent` here?

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/Event.java [hunk 3]
```diff
@@ -1,10 +1,12 @@
 package uk.gov.hmcts.reform.bulkscanprocessor.entity;
 
+// TODO: separate envelope state and events.
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/Event.java]: #70 Done as per conversation the other day

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/EnvelopeNotFoundException.java [hunk 4]
```diff
@@ -0,0 +1,4 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.exceptions;
+
+public class EnvelopeNotFoundException extends RuntimeException {
+}
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/EnvelopeNotFoundException.java]: Missing entry in `ControllerAdvice` to properly respond with 404

-- EDIT --

Ah, service is not actually plugged in anywhere yet. Can be done when plugged in controller ;)

**Pull request merged**